### PR TITLE
perf: Replacing artwork(id) with artworks query

### PR DESCRIPTION
**Pull request by rajsam003**
This PR improves the performance of the app by fetching all artworks at once instead of nesting query renderers for the albums. 

Before: 
```
artwork(id: $id) {
      ...ArtworkGridItem_artwork
    }

```

After:
```
artworks(ids: [id1, id2, id3, ...]) {
      edges{
           node{
          ...ArtworkGridItem_artwork
          }
       }
    }
```

### PR Checklist

- [x] I tested my changes on **iOS** and **Android**.
- [ ] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

Need help with something? Have a look at our [docs], or get in touch with us.

[docs]: /docs/README.md

### CODE REVIEW

File: src/app/screens/HomeTabs/Albums/CreateOrEditAlbum/CreateOrEditAlbum.tsx [hunk 1]
```diff
@@ -44,6 +48,18 @@ export const CreateOrEditAlbum = () => {
...
+    To avoid that we are giving a condition to only call the API when the selectedArtworks is not empty*/
+  const artworks =
+    selectedArtworks.length > 0
+      ? extractNodes(
+          useLazyLoadQuery<AlbumArtworksQuery>(albumArtworksQuery, {
+            artworkIds: selectedArtworks,
+          }).artworks
+        )
+      : []
+
```
Comment (rajsam003) [inline src/app/screens/HomeTabs/Albums/CreateOrEditAlbum/CreateOrEditAlbum.tsx:62]: @mdole Here is the case I was talking about☝️

File: src/app/screens/HomeTabs/Albums/CreateOrEditAlbum/CreateOrEditAlbum.tsx [hunk 2]
```diff
@@ -44,6 +48,18 @@ export const CreateOrEditAlbum = () => {
   const selectedArtworks = uniq([...(album?.artworkIds || []), ...selectedArtworksInModel])
   const space = useSpace()
 
+  /* If we pass empty ids(selectedArtworks) as parameter in the albumArtworksQuery API,
```
Comment (MounirDhahri) [inline src/app/screens/HomeTabs/Albums/CreateOrEditAlbum/CreateOrEditAlbum.tsx:51]: **praise:** thanks for the comment, it makes it easier to understand your reasoning here.

Comment (MounirDhahri) [reply src/app/screens/HomeTabs/Albums/CreateOrEditAlbum/CreateOrEditAlbum.tsx:51]: **suggestion:** I am wondering though if a hotfix is what we need here, I would suggest here to add a new argument to the interface (`allowEmptyReturns` or something like that) to let it know to return nothing if nothing is provided. This way we can fix the problem everywhere in Artsy and make the next dev life a bit easier

Speaking of making a change here https://www.notion.so/artsy/Sprint-17-cdd05a12330e4b27b9c2ba16a8a9d812

**Review by MounirDhahri — COMMENTED** [review]
Nice PR Sam, thanks for making this change ❤️ I left a small comment, let me know if you didn't get what I mean

File: src/app/screens/HomeTabs/Albums/CreateOrEditAlbum/CreateOrEditAlbum.tsx [hunk 3]
```diff
@@ -44,6 +48,18 @@ export const CreateOrEditAlbum = () => {
...
+    it returns random artworks(instead of returning empty object),
```
Comment (patrinoua) [inline src/app/screens/HomeTabs/Albums/CreateOrEditAlbum/CreateOrEditAlbum.tsx:52]: Wow this sounds bad! Thoughts @brainbicycle @mc-jones @lidimayra ?

Comment (rajsam003) [reply src/app/screens/HomeTabs/Albums/CreateOrEditAlbum/CreateOrEditAlbum.tsx:52]: We already have a thread for this
https://artsy.slack.com/archives/CP9P4KR35/p1661419259015689

Comment (brainbicycle) [reply src/app/screens/HomeTabs/Albums/CreateOrEditAlbum/CreateOrEditAlbum.tsx:52]: hmmm, it sounds like weird behavior by the api to me but also if we don't have artwork ids I think it makes sense not to query the api at all? so the frontend guard against that condition seems like enough to me

Comment (brainbicycle) [reply src/app/screens/HomeTabs/Albums/CreateOrEditAlbum/CreateOrEditAlbum.tsx:52]: but I actually think the bigger concern from that thread is the endpoint is deprecated so we may not want to use this at all in folio?

**Review by brainbicycle — CHANGES_REQUESTED** [review]
Thanks Sam for thinking about the performance here! I think unfortunately we may want to keep it as is for now because from the thread it sounds like this endpoint is deprecated. It is possible we could use artworksConnection or filterArworksConnection for the same purpose but I would suggest holding off until we have a better sense of what offline access looks like. Reason being if offline ends up requiring fetching all data up front that might dictate which endpoints we use and also would mean that once fetched going to a page would be instant.

**Comment by rajsam003** [comment]
> Thanks Sam for thinking about the performance here! I think unfortunately we may want to keep it as is for now because from the thread it sounds like this endpoint is deprecated. It is possible we could use artworksConnection or filterArworksConnection for the same purpose but I would suggest holding off until we have a better sense of what offline access looks like. Reason being if offline ends up requiring fetching all data up front that might dictate which endpoints we use and also would mean that once fetched going to a page would be instant.

Makes total sense Brian

**Comment by rajsam003** [comment]
> Should we close this pr?

Ya sure. I am closing it.

**Pull request closed without merging**